test/0.8.9/utils/pausable-util.test.sol

```diff
@@ -0,0 +1,186 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.9;
+
+import "forge-std/Test.sol";
+import { PausableUntil } from "contracts/0.8.9/utils/PausableUntil.sol";
+
+contract ExposedPausableUntil is PausableUntil {
+    function setPausedState(uint256 _resumeSince) public {
+        _setPausedState(_resumeSince);
+    }
+
+    function pauseUntil(uint256 _pauseUntilInclusive) public {
+        _pauseUntil(_pauseUntilInclusive);
+    }
+
+    function pauseFor(uint256 _duration) public {
+        _pauseFor(_duration);
+    }
+
+    function resume() public {
+        _resume();
+    }
+
+    function checkResumed() public view {
+        _checkResumed();
+    }
+
+    function checkPaused() public view {
+        _checkPaused();
+    }
+}
+
+contract PausableUntilTest is Test {
+    ExposedPausableUntil public pausableUntil;
+
+    function setUp() public {
+        pausableUntil = new ExposedPausableUntil();
+    }
+
+    function testIsPaused() public {
+        assertEq(pausableUntil.isPaused(), false);
+    }
+
+    function testSetPausedState(uint256 _resumeSince, uint256 _random_time) public {
+        // ignore obvious 1 state and underflow condition
+        vm.assume(_resumeSince != 1);
+        vm.assume(_resumeSince >= block.timestamp);
+
+        // make sure we're not already paused
+        assertEq(pausableUntil.isPaused(), false);
+
+        // pause until _resumeSince
+        pausableUntil.setPausedState(_resumeSince);
+        assertEq(pausableUntil.isPaused(), true);
+
+        // check a random point in time (either before or after _resumeSince)
+        vm.warp(_random_time);
+        if (_random_time < _resumeSince) {
+            assertEq(pausableUntil.isPaused(), true);
+        } else {
+            assertEq(pausableUntil.isPaused(), false);  
+        }
+
+        // go forward in time to _resumeSince - 1 block, check edge
+        vm.warp(_resumeSince - 1);
+        assertEq(pausableUntil.isPaused(), true);
+
+        // go forward in time to _resumeSince block, check edge
+        vm.warp(_resumeSince);
+        assertEq(pausableUntil.isPaused(), false);
+    }
+
+    function testPauseUntil(uint256 _pauseUntilInclusive, uint256 _random_time) public {
+        vm.assume(_pauseUntilInclusive != 0);
+        vm.assume(_random_time != 0);
+        vm.assume(_random_time != _pauseUntilInclusive);
+                
+        if (_pauseUntilInclusive < block.timestamp) {
+            vm.expectRevert();
+            pausableUntil.pauseUntil(_pauseUntilInclusive);
+            assertEq(pausableUntil.isPaused(), false);
+        } else {
+            pausableUntil.pauseUntil(_pauseUntilInclusive);
+            assertEq(pausableUntil.isPaused(), true);
+        }
+
+        // check a random point in time (either before or after _pauseUntilInclusive)
```

Comment:
can we add additional test/assert for PAUSE_INFINITELY value for `pauseFor` and `pauseUntil`?

Reply:
Yeah, agree :+1: